src/scripts/main.js

```diff
@@ -1,3 +1,273 @@
+
+    addTile();
+    addTile();
+  } else if (e.target.classList.contains('restart')) {
+    restart();
+    addTile();
+    addTile();
+  }
+});
+
+function moveUp(index) {
+  let colNums = [];
+
+  rows.forEach(row => {
+    colNums.push(+row.cells[index].textContent);
+  });
+
+  colNums = colNums.filter(num => !!num);
+
+  for (let i = 0; i < colNums.length - 1; i++) {
+    if (colNums[i] === colNums[i + 1]) {
+      score.textContent = +score.textContent + colNums[i] * 2;
+      colNums[i] *= 2;
+      colNums[i + 1] = 0;
+    }
+  }
+
+  colNums = colNums.filter(num => !!num);
+
+  while (colNums.length < 4) {
+    colNums.push(0);
+  }
+
+  rows.forEach((row, i) => {
+    row.cells[index].textContent = colNums[i] || '';
+  });
+  addCorrectClasses();
+}
+
+function moveDown(index) {
+  let colNums = [];
+
+  rows.forEach(row => {
+    colNums.push(+row.cells[index].textContent);
+  });
+
+  colNums = colNums.reverse().filter(num => !!num);
+
+  for (let i = 0; i < colNums.length - 1; i++) {
+    if (colNums[i] === colNums[i + 1]) {
+      score.textContent = +score.textContent + colNums[i] * 2;
+      colNums[i] *= 2;
+      colNums[i + 1] = 0;
+    }
+  }
+
+  colNums = colNums.reverse().filter(num => !!num);
+
+  while (colNums.length < 4) {
+    colNums.unshift(0);
+  }
+
+  rows.forEach((row, i) => {
+    row.cells[index].textContent = colNums[i] || '';
+  });
+  addCorrectClasses();
+}
+
+function moveRight() {
+  rows.forEach(row => {
+    let rowNums = [];
+
+    [...row.cells].forEach(cell => rowNums.push(+cell.innerHTML));
+
+    rowNums = rowNums.reverse().filter(num => !!num);
+
+    for (let i = 0; i < rowNums.length - 1; i++) {
+      if (rowNums[i] === rowNums[i + 1]) {
+        score.textContent = +score.textContent + rowNums[i] * 2;
+        rowNums[i] *= 2;
+        rowNums[i + 1] = 0;
+      }
+    }
+
+    rowNums = rowNums.reverse().filter(num => !!num);
+
+    while (rowNums.length < 4) {
+      rowNums.unshift(0);
+    }
+
+    [...row.cells].forEach((cell, i) => {
+      cell.textContent = rowNums[i] || '';
+    });
+  });
+  addCorrectClasses();
+}
+
+function moveLeft() {
+  rows.forEach(row => {
+    let rowNums = [];
+
+    [...row.cells].forEach(cell => rowNums.push(+cell.innerHTML));
+
+    rowNums = rowNums.filter(num => !!num);
+
+    for (let i = 0; i < rowNums.length - 1; i++) {
+      if (rowNums[i] === rowNums[i + 1]) {
+        score.textContent = +score.textContent + rowNums[i] * 2;
+        rowNums[i] *= 2;
+        rowNums[i + 1] = 0;
+      }
+    }
+
+    rowNums = rowNums.filter(num => !!num);
+
+    while (rowNums.length < 4) {
+      rowNums.push(0);
+    }
+
+    [...row.cells].forEach((cell, i) => {
+      cell.textContent = rowNums[i] || '';
+    });
+  });
+  addCorrectClasses();
+}
+
+function addCorrectClasses() {
+  for (const cell of cells) {
+    cell.className = 'field-cell';
+
+    switch (cell.textContent) {
+      case '2': cell.classList.add('field-cell--2'); break;
```

Comment:
Can be replaced with:

![image](https://user-images.githubusercontent.com/56752562/236617169-8cea56ad-34be-40c6-99e9-6d9ce3ee82a1.png)